src/util-base64.c

```diff
@@ -394,6 +400,29 @@ static int B64DecodeStringEndingSpaces(void)
     PASS;
 }
 
+static int B64DecodeStrictModeInvalidPadding(void)
+{
+    const char *src = "D=aB";
```

Comment:
What about `====` ?

Reply:
And `D=` ?

Reply:
And `X===` ?

Reply:
`====` -> should probably be rejected.
`D=` and `X===` -> seems like a valid base64 encoded string.
wdyt?

> Fuzzer could check that successful strict base64 decoding, can be reencoded in the same data as the input

ok. Will extend it. Creating a ticket for this one. Thanks!

Reply:
`D=` is not the padding supposed to go all the way to have the base64 data length a multiple of 4 ?

`X===` : I have only 6 bits, I do not have enough for one byte

Reply:
> D= is not the padding supposed to go all the way to have the base64 data length a multiple of 4 ?

No. Currently, in our decoder, as long as there is space in destination buffer, we will add padding bytes of our own and try to decode it under RFC 4648 (incl strict mode). JFYI that's how popular decoders also seem to do that.

> X=== : I have only 6 bits, I do not have enough for one byte

Sure. That goes for all kinds of padding and the 6 bits with two 00 at the end is considered one byte in such a case. This is also accepted by popular decoders.

Lmk what you think.

Reply:
My understanding of `strict` mode was that it accepts only input that we would be output of my `base64` command line, and `echo -n -e '\x00' | base64` is `AA==`

Reply:
I see. Well, that is not the current behavior. I can make that the behavior though.
Please note that the strict mode is used by datasets only.
Shall I make the strict mode in accordance with the `base64` utility's output?

Reply:
What makes the `base64` command the ultimate source of truth?

Reply:
Well.. it seemed to me that Philippe intended to say it tends to reject more strictly so that must be the strict mode. I can't decide what should strict mode mean for us.. or rather for datasets.. 🤔

Reply:
Make sense to me.

Reply:
Side note: http2 rust code has a strict base64 decoder

Reply:
we're we going to add it as an option to the decode_base64 transform?

Reply:
Just move util-base64.c to some rust base64.rs

Reply:
>  we're we going to add it as an option to the decode_base64 transform?

had forgotten about the `decode_base64` transform altogether. Yes, it takes all the modes as an option indeed.

Reply:
Thanks both for bringing together all the necessary info that is needed to move forward.